Ports/cavestory/package.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port='cavestory'
+version='2.6.5-1'
+files=(
+    "https://github.com/nxengine/nxengine-evo/archive/refs/tags/v${version}.zip v${version}.zip a3a1a802eda46f9922065a35624ba94e3823649a7e71012e479ed91fcdb50255"
+    'https://www.cavestory.org/downloads/cavestoryen.zip cavestoryen.zip aa87fa30bee9b4980640c7e104791354e0f1f6411ee0d45a70af70046aa0685f'
+)
+auth_type='sha256'
+depends=(
+    'libjpeg'
+    'libpng'
+    'SDL2'
+    'SDL2_image'
+    'SDL2_mixer'
+    'SDL2_ttf'
+)
+workdir="nxengine-evo-${version}"
+launcher_name='Cave Story'
+launcher_category='Games'
+launcher_command='/usr/local/bin/nxengine-evo'
+icon_file='platform/switch/icon.jpg'
+
+post_fetch () {
+  mkdir -p ${workdir}/build
+}
+
+build () {
+  cd ${workdir}/build
+  cmake -DCMAKE_BUILD_TYPE=Release .. 
+  make
+}
+
+pre_install () {
+  mv CaveStory/Doukutsu.exe ${workdir}/
+  mv CaveStory/data/* ${workdir}/data/
+  rm -rf CaveStory
```

Comment:
Why are we deleting this data?

Reply:
The "rm -rf CaveStory" is there to delete the unzipped folder, now that all of the game's data was extracted from it. Does Serenity automatically clean this folder up? If not there would just be a folder sitting there taking up space.

Reply:
That's only one folder though. We download tarballs and extract all sorts of files as part of the Toolchain build, project build, and every port build. None of it is cleaned up unless you take action yourself. And it's all downloaded into the Build/<arch> folder anyway 

What if I run the build and install steps again? It will fail unless I do a full clean rebuild of the port.

Reply:
You're right, but I feel it's best to not have unnecessary files, even if other ports do it.

Reply:
For reproducibility and debugging purposes, it's best to leave the files after running the steps. This is the same for all ports, so we should not diverge here.

If anyone's concerned with disk space usage, an option could be trivially added to the invocation of `package.sh` to clean up afterwards (but let's not do that in this PR).